vpr/src/draw/draw_rr.cpp

```diff
@@ -291,228 +251,101 @@ void draw_rr_edges(RRNodeId inode, ezgl::renderer* g) {
-
-                    case e_rr_type::CHANX:
-                        if (rgb_is_same(draw_state->draw_rr_node[inode].color, ezgl::MAGENTA)) {
-                            ezgl::color color = draw_state->draw_rr_node[to_node].color;
-                            g->set_color(color, transparency_factor);
-                        } else if (rgb_is_same(draw_state->draw_rr_node[to_node].color, ezgl::MAGENTA)) {
-                            ezgl::color color = draw_state->draw_rr_node[inode].color;
-                            g->set_color(color, transparency_factor);
-                        } else if (!edge_configurable) {
-                            ezgl::color color = blk_DARKGREY;
-                            g->set_color(color, transparency_factor);
-                        } else {
-                            g->set_color(blk_DARKGREEN, transparency_factor);
-                        }
-                        switch_type = rr_graph.edge_switch(rr_node, iedge);
-                        draw_chanx_to_chanx_edge(rr_node, RRNodeId(to_node),
-                                                 switch_type, g);
-                        break;
-
-                    case e_rr_type::CHANY:
-                        if (rgb_is_same(draw_state->draw_rr_node[inode].color, ezgl::MAGENTA)) {
-                            ezgl::color color = draw_state->draw_rr_node[to_node].color;
-                            g->set_color(color, transparency_factor);
-                        } else if (rgb_is_same(draw_state->draw_rr_node[to_node].color, ezgl::MAGENTA)) {
-                            ezgl::color color = draw_state->draw_rr_node[inode].color;
-                            g->set_color(color, transparency_factor);
-                        } else if (!edge_configurable) {
-                            g->set_color(blk_DARKGREY, transparency_factor);
-                        } else {
-                            g->set_color(blk_DARKGREEN, transparency_factor);
-                        }
-                        switch_type = rr_graph.edge_switch(rr_node, iedge);
-                        draw_chanx_to_chany_edge(inode, to_node,
-                                                 FROM_X_TO_Y, switch_type, g);
-                        break;
-
-                    default:
-                        vpr_throw(VPR_ERROR_OTHER, __FILE__, __LINE__,
-                                  "in draw_rr_edges: node %d (type: %d) connects to node %d (type: %d).\n",
-                                  inode, from_type, to_node, to_type);
-                        break;
+            case e_rr_type::CHANX:
+            case e_rr_type::CHANY:
+                if (to_type == e_rr_type::IPIN) {
+                    color = blk_LIGHTSKYBLUE;
+                    if (draw_state->draw_rr_node[to_node].node_highlighted && draw_state->draw_rr_node[inode].color == DEFAULT_RR_NODE_COLOR) {
+                        // If the IPIN is clicked on, draw connection to all the CHANX
```

Comment:
I think the overall highlighting code needs to be tested and this comment needs to be hoisted higher and updated.
I believe you can right click on an rr-node, and its fanin and fanout  will be highlighted in certain colours.  The highlighted node itself is drawn in magenta. You should test exactly what happens now, and see if you like it.  I think the code can be simplified (which you've done) but you need to test we're doing some reasonable highlighting (idea: right click highlights and rr-node and its fan-in edges and fan-out edges).

Reply:
Maybe the highlighting can be simplified to something that just looks at the state of the prior and current rr-nodes (without paying attention to their type) and decides if an edge should be highlighted or not.  I suspect we're being overly complex with the code that checks the node types (CHANX etc.).  If we can simplify, we could just move it to a helper function to find the colour.